### PR TITLE
Increase Equihash parameters to n = 96, k = 3 (about 430 MiB)

### PR DESCRIPTION
**Pull request by str4d**
Includes a tweak to set the fixed-width of the rows based on whether the first
or last round is widest. This is necessary for some parameters, these ones
included.

### CODE REVIEW

**Comment by str4d** [comment]
This is a stepping stone towards our target parameters, that will enable us to see how the higher-memory slower PoW behaves in combination with the new difficulty algorithm (#931).

Requires regenerating `miner_tests`, which I will do when it is this PR's turn to be merged :)

**Comment by defuse** [comment]
ACK

**Comment by str4d** [comment]
Rebased on latest, and miner tests updated :smile:

File: src/crypto/equihash.h [hunk 1]
```diff
@@ -108,6 +108,8 @@ class TruncatedStepRow : public StepRow<WIDTH>
     eh_trunc* GetTruncatedIndices(size_t len, size_t lenIndices) const;
 };
 
+inline constexpr const size_t max(const size_t A, const size_t B) { return A > B ? A : B; }
```
Comment (ebfull) [inline src/crypto/equihash.h]: Can we use `std::max` instead?

Comment (str4d) [inline src/crypto/equihash.h]: Not in C++ 11, because `std::max` is only `const`, not `constexpr`. C++ 14 fixes this.

**Comment by ebfull** [comment]
ACK

@zkbot r+

**Comment by daira** [comment]
I intend to do a post-hoc review of this today, Internet access permitting.

**Comment by daira** [comment]
ACK. Editing commented-out code (in the tests) is error-prone though. Can we delete that code?

**Comment by str4d** [comment]
We are using that code to regenerate the miner tests, so it is uncommented when being edited. Once we no longer need to regenerate those tests, we could remove that code.

**Pull request merged**